hy/_compat.py

```diff
@@ -14,3 +15,33 @@ def reraise(exc_type, value, traceback=None):
         raise value.with_traceback(traceback)
     finally:
         traceback = None
+
+
+if PY3_9:
+
+    import ast, math, cmath
+    class Unparser(ast._Unparser):
+      # Work around some limitations of Python's `ast.unparse`.
+
+        def visit_Constant(self, node):
+            if isinstance(node.value, (float, complex)):
+                self.write(repr(node.value)
+                    .replace("inf", ast._INFSTR)
+                    .replace('nan', '({}-{})'.format(ast._INFSTR, ast._INFSTR)))
+            else:
+                super().visit_Constant(node)
+
+        def visit_Set(self, node):
+            if node.elts:
+                super().visit_Set(node)
+            else:
+                self.write('{1}.__class__()')
```

Comment:
(trivial, but) is there any significant difference between the representation here vs in your upstream patch `{*()}`?

Reply:
They should be equivalent. I've changed it to `{*()}`.